pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -129,6 +139,18 @@ func (t ResolvedPipelineRunTask) IsCancelled() bool {
 	return c.IsFalse() && c.Reason == v1alpha1.TaskRunSpecStatusCancelled
 }
 
+// isSkippedAsStateConflicted return true only if the taskRun has failed with reason SkippedAsStateConflicted
```

Comment:
I think it would be semantically more correct to actually skip the taskrun, and not fail it with a "skipped" status.

Reply:
we need a way to signify that a parent task was skipped and hence its children should be skipped too and that the task is visited and was skipped